rpc/events.go

```diff
@@ -8,10 +8,10 @@ type EventsArg struct {
 }
 
 type EventFilter struct {
-	FromBlock *BlockID     `json:"from_block"`
-	ToBlock   *BlockID     `json:"to_block"`
-	Address   *felt.Felt   `json:"address"`
-	Keys      []*felt.Felt `json:"keys"`
+	FromBlock *BlockID      `json:"from_block"`
+	ToBlock   *BlockID      `json:"to_block"`
+	Address   *felt.Felt    `json:"address"`
```

Comment:
Why are they, not value type? Surely someone could pass null for `FromBlock`, `ToBlock` and `Address` and there is no `omitempty` tag so they will always be present in EventFilter struct.

Reply:
Also, `BlockID.Hash` should not be a pointer either

Reply:
They are optional, so they should be pointers

Reply:
According to the spec, nothing is a [required](https://github.com/starkware-libs/starknet-specs/blob/aaea417f193bbec87b59455128d4b09a06876c28/api/starknet_api_openrpc.json#L889) for EventFilter. So shouldn't EventFilter look like the following:

```go
type EventFilter struct {
	FromBlock *BlockID      `json:"from_block"`
	ToBlock   *BlockID      `json:"to_block"`
	Address   *felt.Felt    `json:"address"`
    Keys      *[][]felt.Felt `json:"keys"`
}
```

Reply:
I just realised that since slices are reference types there is no need to have `*` in front of `[][]felt.Felt`. I became confused between this and when we want to differentiate between the zero value of the type vs the missing value.